src/code/z_demo.c

```diff
@@ -1,99 +1,1638 @@
+        D_801BB160 = CS_TEXTBOX_TYPE_LEARN_SONG;
+        D_801BB128 = cmd->base;
+        func_80152434(globalCtx, cmd->base);
+    } else {
+    else_label:
+        if (csCtx->frames >= cmd->endFrame) {
+            originalCsFrames = csCtx->frames;
+            dialogState = Message_GetState(&globalCtx->msgCtx);
+            if (dialogState != 2 && dialogState != 0 && dialogState != 7 && dialogState != 8) {
+                csCtx->frames--;
+                if (dialogState == 4 && func_80147624(globalCtx)) {
+                    if (globalCtx->msgCtx.choiceIndex == 0) {
+                        if (cmd->base == 0x33BD) {
+                            func_8019F230();
+                        }
+
+                        if (cmd->textId1 != 0xFFFF) {
+                            func_80151938(globalCtx, cmd->textId1);
+                            if (cmd->type == CS_TEXTBOX_TYPE_3) {
+                                D_801BB160 = CS_TEXTBOX_TYPE_3;
+                                if (cmd->textId2 != 0xFFFF) {
+                                    csCtx->frames++;
+                                }
+                            }
+                        } else {
+                            func_801477B4(globalCtx);
+                            csCtx->frames++;
+                        }
+                    } else {
+                        if (cmd->base == 0x33BD) {
+                            func_8019F208();
+                        }
+
+                        if (cmd->textId2 != 0xFFFF) {
+                            func_80151938(globalCtx, cmd->textId2);
+                            if (cmd->type == CS_TEXTBOX_TYPE_3) {
+                                D_801BB160 = CS_TEXTBOX_TYPE_3;
+                                if (cmd->textId1 != 0xFFFF) {
+                                    csCtx->frames++;
+                                }
+                            }
+                        } else {
+                            func_801477B4(globalCtx);
+                            csCtx->frames++;
+                        }
+                    }
+                }
+
+                if (dialogState == 5 && func_80147624(globalCtx) != 0) {
+                    func_80152434(globalCtx, cmd->base);
+                }
+            }
+
+            if ((dialogState == 2) && (D_801BB160 == CS_TEXTBOX_TYPE_3)) {
+                csCtx->frames--;
+                D_801BB124++;
+            }
+
+            if (originalCsFrames == csCtx->frames) {
+                Interface_ChangeAlpha(1);
+                D_801BB124 = 0;
+                D_801BB128 = 0;
+                func_80161C0C();
+            } else {
+                func_80161BE0(1);
+            }
+        }
+    }
+}
+
+// Related to actorActions. Maybe a generic actorAction setter?
+void func_800ECD7C(CutsceneContext* csCtx, u8** cutscenePtr, s16 index) {
+    s32 i;
+    s32 count;
+
+    bcopy(*cutscenePtr, &count, sizeof(s32));
+    *cutscenePtr += sizeof(s32);
```

Comment:
You were going to change this to `sizeof(count)`, I think? And possibly others elsewhere.